deps/rabbit/test/feature_flags_SUITE.erl

```diff
@@ -112,59 +112,90 @@ end_per_suite(Config) ->
 init_per_group(feature_flags_v1, Config) ->
     rabbit_ct_helpers:set_config(Config, {enable_feature_flags_v2, false});
 init_per_group(feature_flags_v2, Config) ->
-    rabbit_ct_helpers:set_config(Config, {enable_feature_flags_v2, true});
+    %% Before we run `feature_flags_v2'-related tests, we must ensure that both
+    %% umbrellas support them. Otherwise there is no point in running them. The
+    %% `feature_flags_v1' group already covers testing in that case.
+
+    %% To determine that `feature_flags_v2' are supported, we can't
+    %% query RabbitMQ which is not started. Therefore, we check if the
+    %% source or bytecode of `rabbit_ff_controller' is present.
+
+    IsSupported = case rabbit_ct_helpers:get_config(Config, rabbitmq_run_cmd) of
+                      undefined ->
+                          %% make
+                          Dir1 = ?config(rabbit_srcdir, Config),
+                          File1 = filename:join([Dir1, "src", "rabbit_ff_controller.erl"]),
+                          SupportedPrimary = filelib:is_file(File1),
+                          case rabbit_ct_helpers:get_config(Config, secondary_umbrella) of
```

Comment:
The order of checks was reversed in this patch.

The use of a secondary umbrella should be checked first. If there is none configured, the `feature_flags_v2` test group should be executed regardless of the support. Because the test group was committed with the code, it's an error if the test group runs without the corresponding code in a non-mixed-version scenario.